nepomuk/src/main/java/edu/asu/diging/gilesecosystem/nepomuk/core/service/impl/RequestProcessor.java

```diff
@@ -126,6 +138,31 @@ public void processRequest(IStorageRequest request) {
         }
     }
     
+    @Override
+    public void processRequest(IStorageDeletionRequest request) {
+        List<IFile> files = filesManager.getFilesByDocumentId(request.getDocumentId());
+        for (IFile file : files) {
+            IFileTypeHandler handler = fileHandlerRegistry.getHandler(file.getFileType());
+            handler.deleteFile(file);
+        }
+        if (filesManager.getFilesByDocumentId(request.getDocumentId()).isEmpty()) {
```

Comment:
why is this `if` here?

Reply:
Added the if condition to ensure all the files of the document are deleted before sending a deletion complete request to giles

Reply:
why wouldn't it be though? They would only not be empty if the deletion didn't work and in that case there should be an exception or something that could be caught?